### DISCUSSION TITLE
Please consider raising the image resolution & file size limits

### DISCUSSION THREAD

**Discussion by Novaenia** [bluesky-social/atproto#2887] [Bluesky Lexicons]
There's been a large influx of users, mostly artists, coming from Twitter due to recent crappy changes to their terms of service (Using art to train AI slop even without consent)

Unfortunately, it seems like the limits for images on Bluesky are even worse than Twitter's. 

Twitter: The max dimensions are 4096x4096, images larger get resized. Uploaded PNGs are converted to a JPG with subpar quality but for artists there's actually some wiggle room for PNGs, and JPGs are _preserved as-is_ if they meet certain criteria (less than 4096x4096 and  5 MB) This means uploading your own 100% quality JPG is the best choice on Twitter if using a PNG would get it compressed by their servers. There's a [thread going into detail here](https://devcommunity.x.com/t/118695).

Correct me if I'm wrong but from what I've seen:
On Bluesky the maximum dimensions are 2000x2000 and the max file size is 1MB. The server rejects images above 1 MB, so the client compresses the image to a JPG starting at 100% quality, dropping the quality level by 10% until the result is below the size limit.

I think these limits are way too low and should be raised to at least match Twitter's.

**Comment by sroby** [comment]
Seconding this. If BlueSky really wants to convince artists to move over from Twitter, it will need picture quality limits that are _at the very least_ on par with Twitter's, and preferably even better.

**Reply by Tamschi** [reply]
https://github.com/bluesky-social/social-app/pull/6137 should help a lot once merged.
Most JPEGs I see on Twitter are far below 1MB and this PR aims for the best quality within that limit.

Still would be nice to have a bit more room for higher-detail images, though.

**Comment by Novaenia** [comment]
improvements to the uploaded JPEG quality level as @Tamschi mentioned have been merged now, but it would be great if limits were actually raised.

**Reply by Tamschi** [reply]
Uploading certain images as-is (minus e.g. location meta data!) would probably help with storage cost too, since the PDS (iinm) de-deduplicates identical blobs already and different browsers aren't likely to have bit-identical image compression. That would need much more code (implementation work) on the client though, to repackage the file safely, which *may* be the main reason this isn't being done right now.

I wonder what the CDN input limits are vs. the PDS limits. If the former are higher, then in theory better media quality could be a selling point for a third-party PDS provider (once the frontend has an API to get the blob size limit, as [right now it's hard-coded](https://github.com/bluesky-social/social-app/blob/e6c867a961a6dd99de2388ff5d4a3abb5431e55c/src/lib/constants.ts#L97)). The CDN is a major choke point here though, as better image uploads won't help if that quality doesn't reach most viewers.

**Reply by kindgracekind** [reply]
PDS have larger blob limits now (50MB I think?) but you have to upload them via the api. ([example](https://boletus.us-west.host.bsky.network/xrpc/com.atproto.sync.getBlob?did=did:plc:p572wxnsuoogcrhlfrlizlrb&cid=bafkreidh3g7ea4bhqhuer623oxjil7a4fpibwds7dwhzgluykcubl76cke)). I wonder if a "view full size" thing would work, where it bypasses the CDN and fetches directly from the PDS.

**Reply by kindgracekind** [reply]
If there’s interest, I could build a tool to do these uploads and post them with embedded download links. It would be separate from the main client, though.

**Reply by snuggleproxy** [reply]
@kindgracekind I'm def interested in this. How much work would it be?

**Comment by Novaenia** [comment]
bumping this, it would be nice to finally get a word from someone on the Bluesky team about this
